com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs

```diff
@@ -240,6 +242,15 @@ static void ProcessSubGraph(SubGraphAsset asset, GraphData graph)
             {
                 node.CollectShaderProperties(collector, GenerationMode.ForReals);
             }
+            for (int i = 0; i < collector.properties.Count; ++i)
+            {
+                // override nondeterminstic guids with obviously local ids.
+                collector.properties[i].ObjectIdUnsafe = "SubGraphImporter_propertyNode_luid_"+i;
+                byte[] guidBytes = new byte[16];
+                BitConverter.GetBytes(i+1).CopyTo(guidBytes, 0);
+                collector.properties[i].GuidUnsafe = new Guid(guidBytes);
```

Comment:
Not sure how I feel about using the index directly as the GUID.  It should work, but only because we know we're only doing this hack in exactly one spot.  Better would be to hash the objectIdUnsafe string and use that; so at least it depends on the context, not just the index.

If you want to keep it as is, I would add a comment about how this hack is not expected to scale (basically what I said above)

Reply:
There's actually a "standard" way to do this kind of thing for GUIDs:  https://stackoverflow.com/questions/2642141/how-to-create-deterministic-guids
Someone even wrote a C# implementation:
https://github.com/Faithlife/FaithlifeUtility/blob/master/src/Faithlife.Utility/GuidUtility.cs
That's probably overkill though, especially if we think we want to move to a local id system in the future.

Reply:
I'll comment it up more clearly, but I don't really see a point in generating a guid here. Given how we're using them and how they are being generated currently, how would this scale better if a guid was used? If this was a genuinely correct use-case for guids, then yes-- it would, but since it isn't I don't think it matters.